python/drned_xmnr/op/common_op.py

```diff
@@ -71,3 +71,19 @@ def perform(self):
         if self.filter.devcli_error is None:
             return {'success': 'Loaded initial config.'}
         return {'failure': 'Device driver failed.'}
+
+
+class SaveDefaultConfigOp(ActionBase):
+    """ Action handler used to save/create default device configuration
+        by saving running configuration to the filesystem of the device.
+    """
+    action_name = 'xmnr save-default-config'
+
+    def perform(self):
+        try:
+            self.devcli_run('save-default-config.py', [])
+        except BaseException as e:
```

Comment:
I don't think we should catch exceptions here. The `devcli_run` code should not throw them except for `ActionError` (and if it does, it is a symptom of a more serious problem and it will be logged anyway). On the other hand, you should check the return value - 0 is success, anything else means failure; note that essentially, you are spawning a new process here.  (Yes, I see now that load-defailt-config has the same problem, should be fixed too, I guess.)

Reply:
will update to handle result in manner similar to cli2netconf...